server/sonar-server/src/main/java/org/sonar/server/usergroups/ws/SearchAction.java

```diff
@@ -0,0 +1,115 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.usergroups.ws;
+
+import com.google.common.base.Function;
+import com.google.common.collect.Collections2;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import org.apache.commons.lang.StringUtils;
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService.NewController;
+import org.sonar.api.server.ws.WebService.Param;
+import org.sonar.api.utils.text.JsonWriter;
+import org.sonar.core.persistence.DbSession;
+import org.sonar.core.user.GroupDto;
+import org.sonar.server.db.DbClient;
+import org.sonar.server.es.SearchOptions;
+
+public class SearchAction implements UserGroupsWsAction {
+
+  public static final String FIELD_NAME = "name";
+  public static final String FIELD_DESCRIPTION = "description";
+  public static final String FIELD_MEMBERS_COUNT = "membersCount";
+
+  private DbClient dbClient;
+
+  public SearchAction(DbClient dbClient) {
+    this.dbClient = dbClient;
+  }
+
+  @Override
+  public void define(NewController context) {
+    context.createAction("search")
+      .setDescription("Search for user groups")
+      .setHandler(this)
+      .setResponseExample(getClass().getResource("example-search.json"))
+      .setSince("5.2")
+      .addFieldsParam(Arrays.asList(FIELD_NAME, FIELD_DESCRIPTION, FIELD_MEMBERS_COUNT))
+      .addPagingParams(100)
+      .addSearchQuery("sonar-users", "names");
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    int page = request.mandatoryParamAsInt(Param.PAGE);
+    int pageSize = request.mandatoryParamAsInt(Param.PAGE_SIZE);
+    SearchOptions options = new SearchOptions()
+      .setPage(page, pageSize);
+
+    String query = StringUtils.defaultIfBlank(request.param(Param.TEXT_QUERY), "");
+    List<String> fields = request.paramAsStrings(Param.FIELDS);
+
+    DbSession session = dbClient.openSession(false);
+    try {
+      int limit = dbClient.groupDao().countByQuery(session, query);
+      List<GroupDto> groups = dbClient.groupDao().selectByQuery(session, query, options.getOffset(), pageSize);
+      Collection<Long> groupIds = Collections2.transform(groups, new Function<GroupDto, Long>() {
+        @Override
+        public Long apply(@Nonnull GroupDto input) {
+          return input.getId();
+        }
+      });
+      Map<String, Integer> userCountByGroup = dbClient.groupMembershipDao().countUsersByGroups(session, groupIds);
+
+      JsonWriter json = response.newJsonWriter().beginObject();
+      options.writeJson(json, limit);
+      writeGroups(json, groups, userCountByGroup, fields);
+      json.endObject().close();
+    } finally {
+      session.close();
```

Comment:
I prefer to name the variable dbSession, as we're in WS context, it's not obvious what we're talking about. Moreover, why not use the MyBatis.closeQuietly() method ?